locust/main.py

```diff
@@ -229,6 +229,16 @@ def timelimit_stop():
         # spawn web greenlet
         logger.info("Starting web monitor at http://%s:%s" % (options.web_host or "*", options.web_port))
         web_ui = WebUI(environment=environment)
+        if options.web_auth is not None:
+            web_ui.app.config["BASIC_AUTH_ENABLED"] = True
```

Comment:
I think it would be better to add an `auth` argument to `WebUI.__init__` (that defaults to None) and move most of this logic to the WebUI class.

Reply:
Made a change, let me know what you think.